offlineimap/folder/LocalStatusSQLite.py

```diff
@@ -169,7 +182,8 @@ def __sql_write(self, sql, args=None, executemany=False):
                         else:
                             self.connection.execute(sql, args)
                     success = True
-                    self.connection.commit()
+                    if not self._in_transactions:
+                        self.connection.commit()
             except sqlite.OperationalError as e:
```

Comment:
We must honor the configuration option here and in the __enter__/__exit__ special methods.